### PR TITLE
Improve Makefile in Test/tutorial/Simple_compiler

### PR DESCRIPTION
**Pull request by cpitclaudel**
Based on discussions with @seebees.  The idea is to separate the build directory from the source directory and make it easier to draw inspiration from that makefile.

@robin-aws I ran into an issue with xunit for this.  I tried this header in `Compiler.dfy`

```
// RUN: make -C %S DAFNY_COMPILER=%dafny Compiler.dfy.expect
// RUN: %diff "%s.expect" "%t"
```

… but it seems that `%dafny` doesn't get substituted.  Is that on purpose?

### CODE REVIEW

**Comment by seebees** [comment]
You have the `csharp` dir in `.gitignore`. But this places the C# project file at the root.
If there are a lot of runtimes, this is a little cluttered.
I would think that for multipule runtimes you would want a root project for Dafny,
and then a directory for each downstream runtime project.

File: Test/tutorial/Simple_compiler/Makefile [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
 
-$(dll): $(deps)
-	dotnet build
+# Step 3: compile resulting C# project
+csproj := SimpleCompiler.csproj
+csharp_deps := Main.cs Compiler.cs Simple.g4 $(csproj)
+csharp_copied_deps := $(patsubst %,$(csharp_dir)/%,$(csharp_deps))
+csharp_dll := $(csharp_dir)/bin/Debug/net6.0/SimpleCompiler.dll
+$(csharp_dll): $(csharp_copied_deps)
+	dotnet build $(csharp_dir)/$(csproj)
```
[the remark this answers is not in the document]
Comment (cpitclaudel) [reply Test/tutorial/Simple_compiler/Makefile]: It's building the dll

**Comment by cpitclaudel** [comment]
> You have the csharp dir in .gitignore. But this places the C# project file at the root.

How about now?  I moved all C# files to `csharp/` and mvoed the dafny-generated C# files to `csharp/dafny`

**Review by seebees — DISMISSED** [review]
LGTM

**Review by seebees — DISMISSED** [review]
Cool. I would like one day to break out the files, but that does complicate things

File: Test/tutorial/Simple_compiler/Makefile [hunk 2]
```diff
@@ -1,22 +1,40 @@
...
 default: run
 
-Compiler.cs: $(compiler)
-	dotnet run --project $(DafnyDriver).csproj -- \
-		/spillTargetCode:3 /compile:0 /noVerify $<
+# Step 1: Copy shared dependencies
+csharp_dir := csharp
+shared_deps := Simple.g4
+$(csharp_dir)/%: %
+	cp "$<" "$@"
```
[the remark this answers is not in the document]
Comment (cpitclaudel) [reply Test/tutorial/Simple_compiler/Makefile:10]: I did that and tried to get it to work, but it seems that the Antlr4 C# msbuild plugin  doesn't like to read a file from a parent directory (annoyingly)

**Review by keyboardDrummer — DISMISSED** [review]

**Review by atomb — APPROVED** [review]

**Pull request merged**